drivers/counter/counter_mcux_lptmr.c

```diff
@@ -158,99 +159,53 @@ static const struct counter_driver_api mcux_lptmr_driver_api = {
-		.flags = COUNTER_CONFIG_INFO_COUNT_UP,
-		.channels = 0,
-	},
-	.base = (LPTMR_Type *)DT_INST_REG_ADDR(0),
-	.clk_source = TO_LPTMR_CLK_SEL(DT_INST_PROP(0, clk_source)),
-#if DT_INST_NODE_HAS_PROP(0, input_pin)
-#if DT_INST_PROP(0, prescaler) == 1
-	.bypass_prescaler_glitch = true,
-#else
-	.prescaler_glitch = TO_LPTMR_GLITCH(DT_INST_PROP(0, prescaler)),
-#endif
-	.mode = kLPTMR_TimerModePulseCounter,
-	.pin = TO_LPTMR_PIN_SEL(DT_INST_PROP(0, input_pin)),
-	.polarity = DT_INST_PROP(0, active_low),
-#else /* !DT_INST_NODE_HAS_PROP(0, input_pin) */
-	.mode = kLPTMR_TimerModeTimeCounter,
-#if DT_INST_PROP(0, prescaler) == 1
-	.bypass_prescaler_glitch = true,
-#else
-	.prescaler_glitch = TO_LPTMR_PRESCALER(DT_INST_PROP(0, prescaler)),
-#endif
-#endif /* !DT_INST_NODE_HAS_PROP(0, input_pin) */
-	.irq_config_func = mcux_lptmr_irq_config_0,
-};
-
-DEVICE_DT_INST_DEFINE(0, &mcux_lptmr_init, NULL,
-		    &mcux_lptmr_data_0,
-		    &mcux_lptmr_config_0,
-		    POST_KERNEL, CONFIG_COUNTER_INIT_PRIORITY,
-		    &mcux_lptmr_driver_api);
-
-static void mcux_lptmr_irq_config_0(const struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority),
-		    mcux_lptmr_isr, DEVICE_DT_INST_GET(0), 0);
-	irq_enable(DT_INST_IRQN(0));
-}
-#endif	/* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
+#define COUNTER_MCUX_LPTMR_DEVICE_INIT(n)					\
+	static void mcux_lptmr_irq_config_##n(const struct device *dev)		\
+	{									\
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),		\
+			mcux_lptmr_isr, DEVICE_DT_INST_GET(n), 0);		\
+		irq_enable(DT_INST_IRQN(n));					\
+	}									\
+										\
+	static struct mcux_lptmr_data mcux_lptmr_data_##n;			\
+	static void mcux_lptmr_irq_config_##n(const struct device *dev);	\
+										\
+	BUILD_ASSERT(!(DT_INST_PROP(n, lptmr_running_mode) == 1 &&		\
+		DT_INST_PROP(n, prescale_glitch_filter) == 16),			\
+		"Pulse mode cannot have a glitch value of 16");			\
+										\
+	BUILD_ASSERT(DT_INST_PROP(n, resolution) <= 32 &&			\
+		DT_INST_PROP(n, resolution) > 0,				\
+		"LPTMR resolution property should be a width between 0 and 32");\
+										\
+	static struct mcux_lptmr_config mcux_lptmr_config_##n = {		\
+		.info = {							\
+			.max_top_value =					\
+				GENMASK(DT_INST_PROP(n, resolution) - 1, 0),	\
+			.freq = DT_INST_PROP(n, clock_frequency) /		\
+				DT_INST_PROP(n, prescaler),			\
+			.flags = COUNTER_CONFIG_INFO_COUNT_UP,			\
+			.channels = 0,						\
+		},								\
+		.base = (LPTMR_Type *)DT_INST_REG_ADDR(n),			\
+		.clk_source = DT_INST_PROP(n, clk_source),			\
+		.bypass_prescaler_glitch =					\
+			1 - DT_INST_PROP(n, lptmr_running_mode),		\
+		.mode = DT_INST_PROP(n, lptmr_running_mode),			\
+		.pin = DT_INST_PROP_OR(n, input_pin, 0),			\
+		.polarity = DT_INST_PROP(n, active_low),			\
+		.prescaler_glitch = COND_CODE_1(				\
+				DT_INST_PROP(n, lptmr_running_mode),		\
+				(DT_INST_PROP(n, prescale_glitch_filter)),	\
+				(DT_INST_PROP(n, prescale_glitch_filter) - 1)),	\
```

Comment:
can you just do

`DT_INST_PROP(n, prescale_glitch_filter) + DT_INST_PROP(n, lptmr_running_mode) - 1`

?